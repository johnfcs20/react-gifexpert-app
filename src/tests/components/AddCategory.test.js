
import React from "react";
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory";

describe('Prubeas <addCategory />', () => { 
    
    const  setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    test('Debe de mostrarse correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
     })

     test('Debe de cambiar la caja de texto', () => { 
         const  input = wrapper.find('input');
         const value =  'Hola  Mundo';
         input.simulate('change', {target: {value: value}});
      })

      test('No debe de postear la informacion OnSubmit', () => { 
        wrapper.find('form').simulate('submuit', {});
        expect(setCategories).not.toHaveBeenCalled();
     })

     test('Debe de llamar al SetCategory y llenar la caja de texto', () => { 
        const value =  'Hola  Mundo';
        wrapper.find('input').simulate('change', {target: {value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
      })

 })