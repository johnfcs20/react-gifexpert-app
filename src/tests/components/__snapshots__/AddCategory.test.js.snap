// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubeas <addCategory /> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
